app/js/components/cohorts/actionsComponent.jsx

```diff
@@ -0,0 +1,215 @@
+import React, {Component, PropTypes} from 'react';
+
+import { ApiHelper } from '../../helpers/apiHelper';
+import { Navigate } from './navigate';
+import CohortTable from './cohortTable';
+import './cohorts.css';
+
+class ActionsComponent extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            allCohort: [],
+            cohortResult: [],
+            toDisplay: [],
+            perPage: 10,
+        };
+
+        this.navigatePage = this.navigatePage.bind(this);
+        this.getPatientsData = this.getPatientsData.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.handleChnage = this.handleChnage.bind(this);
+    }
+
+    componentWillMount() {
+        const apiHelper = new ApiHelper();
+        apiHelper.get('/cohort?v=full')
+            .then(res => {
+                this.setState(Object.assign({}, this.state, {
+                    allCohort: res.results.reverse()
+                }));
+            });
+    }
+
+    handleChnage(e) {
+        e.preventDefault();
+        this.setState({
+            [e.target.name]: e.target.value
+        });
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
+        const cohort = this.state.cohort;
+        if (cohort && !isNaN(cohort) && this.state.hasOwnProperty('name') &&
+            this.state.hasOwnProperty('description')) {
+            const apiHelper = new ApiHelper();
+            apiHelper.post('/cohort', this.getQueryData())
+                .then((res) => {
+                    this.setState(Object.assign({}, this.state, {
+                            allCohort: [res, ...this.state.allCohort]
+                    }));
+                    document.getElementById('cohort').reset();
+                    
+                });
+        } else {
+            this.setState({ error: "all fields are required" });
+        }
+    }
+
+	displayHistory(history, index) {
+		return (
+			<option value={index} key={index}>{history.description}</option>
```

Comment:
for your key use the shortid module, so that you can do something like shortid.generate(). using of array indexes as keys are kind of discouraged